### PR TITLE
[Bugfix] - Fix student statistics reachable scores calculation

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context

- Reachable score was calculated incorrectly
- Definition of reachable points: Reachable points are the points which the student could theoretically already have achieved. This includes points of exercises, where the assessment is already been finished. (-> after the assessmentDueDate)

### Description

- The participation is only relevant for the points achieved by the student and should not be used for the calculation of the reachable score or the max score

### Steps for Testing

1. Log in to Artemis
2. Navigate to the statistics page in the course student view
3. Check that the reachable points are correct (see definition of reachable points above)

### CODE REVIEW

File: src/main/webapp/app/overview/course-score-calculation.service.ts [hunk 1]
```diff
@@ -30,12 +30,18 @@ export class CourseScoreCalculationService {
...
+
             if (isExerciseFinished && isExerciseIncluded) {
                 const maxPointsReachableInExercise = exercise.maxPoints!;
                 if (exercise.includedInOverallScore === IncludedInOverallScore.INCLUDED_COMPLETELY) {
                     maxPointsInCourse += maxPointsReachableInExercise;
                 }
+                // points are reachable if the exercise is released and the assessment is over --> It was possible for the student to get points
+                if (exercise.includedInOverallScore === IncludedInOverallScore.INCLUDED_COMPLETELY && isAssessmentOver) {
+                    reachableMaxPointsInCourse += maxPointsReachableInExercise;
+                }
```
Comment (iozbeyli) [inline src/main/webapp/app/overview/course-score-calculation.service.ts:44]: Something small but this can be added to the if statement above it.

Comment (FuchsDominik) [reply src/main/webapp/app/overview/course-score-calculation.service.ts]: Like that? -> 05f11a4

**Review by airelawaleria — APPROVED** [review]
worked as expected for me

**Review by philgabr — COMMENTED** [review]
In general this works as described.

However, I noticed a difference between the handling of model & text exercise statistics:
-Having created 3 exercises of these types with a) no assessment due date b) assessment due date in the past & c) assessment due date in the future:
 -> modeling: reachable points: 20 != overall points: 30 
 -> text: reachable points: 20 = overall points: 20

**Review by krusche — APPROVED** [review]
Looks good, thanks!

**Comment by krusche** [comment]
> In general this works as described.
> 
> However, I noticed a difference between the handling of model & text exercise statistics:
> -Having created 3 exercises of these types with a) no assessment due date b) assessment due date in the past & c) assessment due date in the future:
> -> modeling: reachable points: 20 != overall points: 30
> -> text: reachable points: 20 = overall points: 20

Just saw this after merging the PR. Are you sure that for the text exercises, all of them have been released? overall points only include released exercises

**Comment by krusche** [comment]
If this is still an issue, we need to fix it in a follow-up PR

**Comment by FuchsDominik** [comment]
> In general this works as described.
> 
> However, I noticed a difference between the handling of model & text exercise statistics:
> -Having created 3 exercises of these types with a) no assessment due date b) assessment due date in the past & c) assessment due date in the future:
> -> modeling: reachable points: 20 != overall points: 30
> -> text: reachable points: 20 = overall points: 20

I am not aware of any differences regarding text- and modeling exercises when it comes to date-handling. If you are sure that for a) the dueDate is set and like @krusche said, the releaseDate in in the past, we can take a look

**Pull request merged**